src/analysis/problematic_exclusions.py

```diff
@@ -0,0 +1,285 @@
+    # from oaklib.resource import OntologyResource
+    # from oaklib.implementations.sqldb.sql_implementation import SqlImplementation
+    # # https://incatools.github.io/ontology-access-kit/introduction.html#basic-python-example
+    # onto_path = os.path.join(MIRROR_DIR, f'{o}.owl')
+    # if not os.path.exists(onto_path):  # temp until can reliably get all ontologies as .owl in mirror/
+    #     onto_path = onto_path.replace('.owl', '.ttl')
+    # resource = OntologyResource(slug=onto_path, local=True)
+    # si = SqlImplementation(resource)
+    # labels = [x for x in si.get_labels_for_curies(list(df_i['term_id']))]
+    # labels = []
+    # for curie in list(df_i['term_id']):
+    #     label = si.get_label_by_curie(curie)
+    #     labels.append(label)
+    """
+    # Get appropriate terms
+    # idealy would use vars from before
+    term_id_field = 'term_id'
+    term_label_field = 'term_label'
+    terms = list(df[term_id_field])
+    terms = [x for x in terms if any([x.startswith(y) for y in prefix_map.keys()])]
+
+    # Instantiate template
+    prefix_sparql_strings = [f'prefix {k}: <{v}>' for k, v in prefix_map.items()]
+    template_obj = Template(template_str)
+    instantiated_str = template_obj.render(
+        prefixes=prefix_sparql_strings,
+        values=' '.join(terms))
+
+    # Basic vars
+    query_template_filename = 'get-labels'
+    onto_filename = os.path.basename(onto_path)
+    results_dirname = onto_filename.replace('/', '-').replace('.', '-')
+    results_dirname = results_dirname + '__' + query_template_filename.replace('.', '-')
+    # noinspection DuplicatedCode
+    results_dirpath = os.path.join(CACHE_DIR, 'robot', results_dirname)
+    results_filename = 'results.csv'
+    command_save_filename = 'command.sh'
+    results_path = os.path.join(results_dirpath, results_filename)
+    command_save_path = os.path.join(results_dirpath, command_save_filename)
+    instantiated_query_path = os.path.join(results_dirpath, 'query.sparql')
+    command_str = f'robot query --input {onto_path} --query {instantiated_query_path} {results_path}'
+
+    # Cache and run
+    os.makedirs(results_dirpath, exist_ok=True)
+    if not (os.path.exists(results_path) and use_cache):
+        with open(instantiated_query_path, 'w') as f:
+            # noinspection PyUnboundLocalVariable
+            f.write(instantiated_str)
+        with open(command_save_path, 'w') as f:
+            f.write(command_str)
+        subprocess.run(command_str.split())
+
+    # Read results and return
+    try:
+        df2 = pd.read_csv(results_path).fillna('')
+    except pd.errors.EmptyDataError:
+        # remove: so that it doesn't read this from cache, though it could be that there were really no results.
+        os.remove(results_path)
+        # could also do `except pd.errors.EmptyDataError as err`, `raise err` or give option for this as a param to func
+        raise RuntimeError('No results found when trying to get labels for terms.')
```

Comment:
@matentzn I got this error when running for ICD10CM. I temporarily worked around this in order to get results for ICD10CM, but for now those results have no labels. See: #65

Reply:
Thanks, I have assigned the issue to myself

Reply:
See #69 will fix it